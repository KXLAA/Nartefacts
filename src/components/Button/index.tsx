import * as React from 'react'

import { StyledButton } from '@/components/Button/styles'
import { ButtonProps } from '@/components/Button/types'

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, forwardedRef) => {
    const { label = 'Button', css, ...buttonProps } = props
    return (
      <StyledButton
        {...buttonProps}
        ref={forwardedRef}
        css={{ ...(css as any) }}
      >
        {label}
      </StyledButton>
    )
  },
)

Button.displayName = 'Button'
