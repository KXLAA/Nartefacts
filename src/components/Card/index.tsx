import Image from 'next/image'

import * as S from '@/components/Card/styles'

export type CardProps = {
  image?: string | null
  colors?: string[] | null
  likeCount?: number | null
  textColor?: string | null
  albumArt?: string | null
  spotify?: string | null
  apple?: string | null
  type?: string | null
  description?: string | null
  title?: string | null
  artist?: {
    name?: string | null
    photoUrl?: string | null
    biography?: string | null
  } | null
}

export const Card: React.FC<CardProps> = ({ albumArt }) => {
  return (
    <S.Card>
      <S.Gradient />
      <S.ImageWrapper>
        <Image
          src={albumArt!}
          height={385}
          width={385}
          alt={'album art'}
          layout="responsive"
        />
      </S.ImageWrapper>
    </S.Card>
  )
}
