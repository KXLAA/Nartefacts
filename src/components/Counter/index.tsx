import * as S from './styles'

export type CounterProps = {
  count?: number | undefined | null
}

export const Counter: React.FC<CounterProps> = ({ count }) => {
  return (
    <S.Wrapper>
      <S.Counter>
        <p>{`${count} Gradients Generated`} </p>
      </S.Counter>
    </S.Wrapper>
  )
}
