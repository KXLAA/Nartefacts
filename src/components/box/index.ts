export { Box } from './Box'
export type { BoxProps } from './box.types'
