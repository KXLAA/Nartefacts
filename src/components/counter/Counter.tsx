import { StyledCounter, StyledWrapper } from './styles'
import { CounterProps } from './types'

export const Counter = ({ count }: CounterProps) => {
  return (
    <StyledWrapper>
      <StyledCounter as="span">
        <span>{`${count || 0} Gradients Generated`} </span>
      </StyledCounter>
    </StyledWrapper>
  )
}
