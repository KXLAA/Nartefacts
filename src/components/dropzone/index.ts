export { Dropzone } from './Dropzone'
export type { UploadState } from './types'
