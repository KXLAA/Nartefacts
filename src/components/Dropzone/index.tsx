import { useCallback } from 'react'
import { useDropzone } from 'react-dropzone'
import * as S from './styles'
import { UploadPlus } from '../Icons'
import { useS3Upload } from 'next-s3-upload'
import {
  useGenerateColorsMutation,
  useUpdateAnalyticsMutation,
} from 'graphql/generated/graphql'
import { RotatingLines } from 'react-loader-spinner'
import { colorsTuple } from 'components/Palette'
import { UploadState } from 'pages/create'

export type DropzoneProps = {
  upload: UploadState
  setUpload: React.Dispatch<React.SetStateAction<UploadState>>
}

export const Dropzone: React.FC<DropzoneProps> = ({ upload, setUpload }) => {
  const { uploadToS3 } = useS3Upload()
  const [generateColors] = useGenerateColorsMutation()
  const [updateAnalytics] = useUpdateAnalyticsMutation()

  const onDrop = useCallback(async (acceptedFiles) => {
    setUpload((prev) => ({ ...prev, isUploading: true }))
    try {
      const { url } = await uploadToS3(acceptedFiles[0])
      const id = '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d'
      setUpload((prev) => ({ ...prev, imageUrl: url }))
      if (url) {
        setUpload((prev) => ({ ...prev, isUploading: false }))
        const { data } = await generateColors({
          variables: {
            imageUrl: url,
          },
        })
        await updateAnalytics({
          variables: {
            id: id,
          },
          optimisticResponse: {
            updateAnalytics: {
              id: id,
              generatedPalettes: 1,
            },
          },
        })
        setUpload((prev) => ({
          ...prev,
          colors: data?.generateColors?.colors as colorsTuple,
        }))
      }
    } catch (error: unknown) {
      if (error instanceof Error) {
        setUpload((prev) => ({ ...prev, error: (error as Error).message }))
      }
    }

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  const { getRootProps, getInputProps, isDragActive } = useDropzone({
    onDrop,
    maxFiles: 1,
    accept: 'image/jpeg,image/png',
  })

  return (
    <S.Drop {...getRootProps()}>
      <input {...getInputProps()} />
      {!upload.error ? (
        <S.Content>
          {upload.isUploading ? (
            <RotatingLines width="100" strokeColor="#FF5733" />
          ) : (
            <>
              <UploadPlus size={72} strokeWidth={0.5} color="#5A5A5A" />

              {isDragActive ? (
                <p>drop the files here ...</p>
              ) : (
                <p>drag {'n'} drop your image here, or click to select files</p>
              )}
            </>
          )}
        </S.Content>
      ) : (
        <p>{upload.error}</p>
      )}
    </S.Drop>
  )
}
