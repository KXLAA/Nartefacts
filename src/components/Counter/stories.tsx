import { Story, Meta } from '@storybook/react'

import { Counter } from '@/components/Counter'

export default {
  component: Counter,
  title: 'Counter',
} as Meta

export const Basic: Story = () => <Counter />
