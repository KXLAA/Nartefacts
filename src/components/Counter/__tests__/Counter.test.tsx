import { render } from '@testing-library/react'

import { Counter } from '@/components/Counter'

describe('<Counter />', () => {
  it('should render the component with passed props', () => {
    const { getByText } = render(<Counter count={4} />)
    const counter = getByText(/4 gradients generated/i)

    expect(counter).toBeInTheDocument()
  })

  it('should render <Counter /> correctly', () => {
    const { container } = render(<Counter />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
