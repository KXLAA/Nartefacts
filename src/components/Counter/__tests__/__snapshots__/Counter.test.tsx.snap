// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> should render <Counter /> correctly 1`] = `
<div
  class="nartefacts-c-AIGlN"
>
  <div
    class="nartefacts-c-XyGMV"
  >
    <p>
      0 Gradients Generated
       
    </p>
  </div>
</div>
`;
